pkg/services/ngalert/notifier/channels/sensugo.go

```diff
@@ -0,0 +1,201 @@
+
+type SensuGoNotifier struct {
+	old_notifiers.NotifierBase
+	log  log.Logger
+	tmpl *template.Template
+
+	URL       string
+	Entity    string
+	Check     string
+	Namespace string
+	Handler   string
+	APIKey    string
+	Message   string
+}
+
+// NewSlackNotifier is the constructor for the Slack notifier
+func NewSensuGoNotifier(model *models.AlertNotification, t *template.Template) (*SensuGoNotifier, error) {
+	if model.Settings == nil {
+		return nil, alerting.ValidationError{Reason: "No Settings Supplied"}
+	}
+
+	url := model.Settings.Get("url").MustString()
+	if url == "" {
+		return nil, alerting.ValidationError{Reason: "Could not find URL property in settings"}
+	}
+
+	apikey := model.DecryptedValue("apikey", model.Settings.Get("apikey").MustString())
+	if apikey == "" {
+		return nil, alerting.ValidationError{Reason: "Could not find the API Key property in settings"}
+	}
+
+	return &SensuGoNotifier{
+		NotifierBase: old_notifiers.NewNotifierBase(model),
+		URL:          url,
+		Entity:       model.Settings.Get("entity").MustString(),
+		Check:        model.Settings.Get("check").MustString(),
+		Namespace:    model.Settings.Get("namespace").MustString(),
+		Handler:      model.Settings.Get("handler").MustString(),
+		APIKey:       apikey,
+		Message:      model.Settings.Get("message").MustString(`{{ template "default.message" .}}`),
+		log:          log.New("alerting.notifier.slack"),
+		tmpl:         t,
+	}, nil
+}
+
+// Notify send alert notification to Sensu Go
+func (sn *SensuGoNotifier) Notify(ctx context.Context, as ...*types.Alert) (bool, error) {
+	sn.log.Info("Sending Sensu Go result")
+
+	data := notify.GetTemplateData(ctx, sn.tmpl, as, gokit_log.NewNopLogger())
+	var tmplErr error
+	tmpl := notify.TmplText(sn.tmpl, data, &tmplErr)
+
+	ruleUIDs := make(map[string]struct{})
+	ruleNames := make(map[string]struct{})
+	for _, a := range as {
+		v, ok := a.Labels[ngmodels.UIDLabel]
+		if ok {
+			_, exists := ruleUIDs[string(v)]
+			if !exists {
+				ruleUIDs[string(v)] = struct{}{}
+			}
+		}
+		v, ok = a.Labels[model.AlertNameLabel]
+		if ok {
+			_, exists := ruleNames[string(v)]
+			if !exists {
+				ruleNames[string(v)] = struct{}{}
+			}
+		}
+	}
+
+	var buffer bytes.Buffer
+	for k := range ruleUIDs {
+		buffer.WriteString(fmt.Sprintf("%s,", k))
+	}
+	strRuleUIDs := strings.TrimRight(buffer.String(), ",")
+
+	buffer.Reset()
+	for k := range ruleNames {
+		buffer.WriteString(fmt.Sprintf("%s,", k))
+	}
+	strRuleNames := strings.TrimRight(buffer.String(), ",")
+
+	// Sensu Go alerts require an entity and a check. We set it to the user-specified
+	// value (optional), else we fallback and use the grafana rule anme  and ruleID.
+	entity := sn.Entity
+	if entity == "" {
+		if ruleNames != nil {
+			// Sensu Go alerts cannot have spaces in them
+			entity = strings.ReplaceAll(strRuleNames, " ", "_")
+		} else {
+			entity = "default"
+		}
+	}
+
+	check := sn.Check
+	if check == "" {
+		if strRuleUIDs != "" {
+			check = strRuleUIDs
```

Comment:
I am not sure if a string of rule IDs would be a right fit for `name` metadata. Since the alert message would get the rule ID as labels, I think we can skip any explicit rule IDs. Here it can be just `default`?

Reply:
This is what we used to do in the [existing](https://github.com/grafana/grafana/blob/32030f3929cfbeaff7ef5db7784b16f1c52f3fe6/pkg/services/alerting/notifiers/sensugo.go#L132) implementation. I don't really have an opinion.